llvm/lib/Target/NVPTX/SYCL/LocalAccessorToSharedMemory.cpp

```diff
@@ -48,7 +48,9 @@ class LocalAccessorToSharedMemory : public ModulePass {
     // Access `nvvm.annotations` to determine which functions are kernel entry
     // points.
     auto NvvmMetadata = M.getNamedMetadata("nvvm.annotations");
-    assert(NvvmMetadata && "IR compiled to PTX must have nvvm.annotations");
+    if (!NvvmMetadata)
```

Comment:
Is this a related change?

Reply:
Yes. We skipped this pass for modules w/o `-sycldevice` triple component in llvm/lib/Target/NVPTX/NVPTXTargetMachine.cpp and now this pass is executed regardless of the environment component value. SYCL compiler always added `nvvm.annotations` metadata, but it might be missing. In case when this metadata is not emitted, we just skip the transformation instead of crashing (with enabled assertions).